src/snowflake/snowpark/session.py

```diff
@@ -248,6 +253,16 @@ def __init__(self, conn: ServerConnection) -> None:
         self._plan_builder = SnowflakePlanBuilder(self)
         self._last_action_id = 0
         self._last_canceled_id = 0
+        self._use_scoped_temp_objects = bool(
+            _use_scoped_temp_objects
+            and (
+                conn._conn._session_parameters.get(
+                    _PYTHON_SNOWPARK_USE_SCOPED_TEMP_OBJECTS_STRING, True
```

Comment:
Does it handle the scenario that this parameter in connection parameters is in lower case? You can just check `conn._lower_case_parameters`. Also, to make code more clear:
```
self._use_scoped_temp_objects = _use_scoped_temp_objects and conn._lower_case_parameters.get(_PYTHON_SNOWPARK_USE_SCOPED_TEMP_OBJECTS_STRING, True)
```
?

Reply:
I think it does because the connection is created from lower case parameters? https://github.com/snowflakedb/snowpark-python/blob/952c9b31efc35b72b7c06f44fb4094a6c778dc81/src/snowflake/snowpark/_internal/server_connection.py#L138. 

I think connection parameters will be set by users. So the user can potentially pass in the lower case parameter (but I'm not sure whether it'll be converted to upper case in the connector).

Reply:
you're right @sfc-gh-jdu , the `_lower_case_parameters` is client-side parameters which contains `schema`, `database` etc.

but it doesn't have the server side session parameter so I'm afraid it can't be used to get the `_PYTHON_SNOWPARK_USE_SCOPED_TEMP_OBJECTS_STRING`

Reply:
I see, I misread it as a client param. Then it should be good